src/js/components/Form/Form.js

```diff
@@ -118,6 +118,12 @@ const Form = forwardRef(
 
             let valid = false;
 
+            setRequiredFields(
```

Comment:
This seems like the wrong place to put this. Why isn't this covered by this block later on?:
```
        if (required) {
          setRequiredFields(prevValue =>
            !prevValue.includes(name) ? [...prevValue, name] : prevValue,
          );
        }
```

Reply:
I can do something like this,

        if (validateArg || required) {
          validations.current[name] = validateField;
          setRequiredFields(prevValue =>
            required 
            && !prevValue.includes(name) 
            && Object.keys(validations.current).includes(name) 
            ? [...prevValue, name] : prevValue,
          );

But in useFormField, on every mount, the requiredFields will be set to empty unless validation triggers.

Reply:
Wanted to make the filter below where you were deleting unmounted components (error/info object keys).
You are right, made the change.